samza-core/src/main/java/org/apache/samza/clustermanager/container/placement/ContainerPlacementUtil.java

```diff
@@ -0,0 +1,273 @@
+ * Key is combination of {@link UUID} and message type (either {@link ContainerPlacementRequestMessage} or {@link ContainerPlacementResponseMessage})
+ * and the value is the actual request or response message
+ */
+public class ContainerPlacementUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ContainerPlacementUtil.class);
+
+  private static final Integer VERSION = 1;
+  public static final String CONTAINER_PLACEMENT_STORE_NAMESPACE = "samza-place-container-v" + VERSION;
+
+  private final NamespaceAwareCoordinatorStreamStore containerPlacementMessageStore;
+  private final ObjectMapper objectMapper = ContainerPlacementMessageObjectMapper.getObjectMapper();
+
+  private boolean stopped = true;
+
+  public ContainerPlacementUtil(MetadataStore containerPlacementMessageStore) {
+    Preconditions.checkNotNull(containerPlacementMessageStore, "MetadataStore cannot be null");
+    this.containerPlacementMessageStore =
+        new NamespaceAwareCoordinatorStreamStore(containerPlacementMessageStore, CONTAINER_PLACEMENT_STORE_NAMESPACE);
+  }
+
+  /**
+   * Perform startup operations. Method is idempotent.
+   */
+  public void start() {
+    if (stopped) {
+      LOG.info("Starting ContainerPlacementStore");
+      containerPlacementMessageStore.init();
+      stopped = false;
+    } else {
+      LOG.warn("already started");
+    }
+  }
+
+  /**
+   * Perform teardown operations. Method is idempotent.
+   */
+  public void stop() {
+    if (!stopped) {
+      LOG.info("stopping");
+      containerPlacementMessageStore.close();
+      stopped = true;
+    } else {
+      LOG.warn("already stopped");
+    }
+  }
+
+  /**
+   * Checks if ContainerPlacementUtil is running
+   */
+  public boolean isRunning() {
+    return !stopped;
+  }
+
+  /**
+   * Writes a {@link ContainerPlacementRequestMessage} to the underlying metastore. This method should be used by external controllers
+   * to issue a request to JobCoordinator
+   *
+   * @param deploymentId identifier of the deployment
+   * @param processorId logical id of the samza container 0,1,2
+   * @param destinationHost host where the container is desired to move
+   * @param requestExpiry optional per request expiry timeout for requests to cluster manager
+   * @param timestamp timestamp of the request
+   * @return uuid generated for the request
+   */
+  public UUID writeContainerPlacementRequestMessage(String deploymentId, String processorId, String destinationHost,
+      Duration requestExpiry, long timestamp) {
+    Preconditions.checkState(!stopped, "Underlying metadata store not available");
+    Preconditions.checkNotNull(deploymentId, "deploymentId should not be null");
+    Preconditions.checkNotNull(processorId, "processorId not be null");
+    Preconditions.checkNotNull(destinationHost, "destinationHost should not be null");
+    Preconditions.checkNotNull(timestamp, "timestamp should not be null");
```

Comment:
Constructor of ContainerPlacementRequestMessage handles null checks, do we need to do another null check here?